addons/web/static/tests/views/search_panel_tests.js

```diff
@@ -4135,5 +4136,34 @@ QUnit.module('Views', {
 
         kanban.destroy();
     });
+
+    QUnit.test("Category with counters and filter with domain", async function (assert) {
+        assert.expect(1);
+
+        const list = await createView({
+            arch: '<tree><field name="foo"/></tree>',
+            archs: {
+                'partner,false,search': `
+                    <search>
+                        <searchpanel>
+                            <field name="category_id" enable_counters="1"/>
+                            <field name="company_id" select="multi" domain="[['category_id', '=', category_id]]"/>
+                        </searchpanel>
+                    </search>`,
+            },
+            data: this.data,
+            model: "partner",
+            services: this.services,
+            View: ListView,
+        });
+
+        assert.strictEqual(
```

Comment:
maybe assert that there are records (sanity check, because otherwise even with counters you'll get that result)